Docs: Blob transfer Architectural Decision Record. (#1259)
eclipse-dataspaceconnector/DataSpaceConnector: docs/developer/decision-records/2022-04-21-dpf-blob-data-transfer/README.md

3. Consumer creates a container where the Provider DPF may write blobs. The container is created only if the client specifies `managedResources=true`.
   The [ObjectStorageProvisioner](../../../../extensions/azure/blobstorage/blob-provision/src/main/java/org/eclipse/dataspaceconnector/provision/azure/blob/ObjectStorageProvisioner.java) is responsible for provisioning the container and for generating a SAS token to access the container. 
   To generate a [SAS token](https://docs.microsoft.com/en-us/azure/storage/common/storage-sas-overview), a storage account key is needed. This storage account key should be stored and retrieved in the Consumer Vault.
4. Consumer stores the SAS token in its Vault.
5. Consumer sends an IDS message to the Provider, containing the information needed to transfer data to the destination container, including the asset id, the destination blob account and container name and the SAS token needed to write a blob to the container.  
6. Provider stores the SAS token in its Vault (rather than in the request to DPF, so that the latter may be persisted in a future implementation without containing secrets).

> 7. Provider initiates the blob transfer on the Provider DPF. The Provider DPF can be embedded or run in a separated runtime. If it runs on a separated runtime, the Provider initiates the transfer via an HTTP request.  

Reviewer: ```suggestion
7. Provider initiates the blob transfer on the Provider DPF. The Provider DPF can be embedded or run in a separated runtime. If it runs on a separated runtime, the Provider's control plane initiates the transfer via an HTTP request.  
```

8. The Provider DPF gets the source storage account access key in the Provider Vault.  

> 9. The Provider DPF gets the SAS token needed to write the blob to the consumer blob container.  

Reviewer: do we really need an access key _and_ a SAS token?

Author: Yes because the `access key` is needed to read the `source blob`.
The `SAS token` is needed to write in the `destination blob`

Reviewer: Uh, you're right of course :)

> 10. The Provider DPF reads the data that needs to be transfered. The [AzureStorageDataSource](../../../../extensions/azure/data-plane/storage/src/main/java/org/eclipse/dataspaceconnector/azure/dataplane/azurestorage/pipeline/AzureStorageDataSource.java) provides the source data stream.  

Reviewer: ```suggestion
10. The Provider DPF reads the data that needs to be transfered using an [AzureStorageDataSource](../../../../extensions/azure/data-plane/storage/src/main/java/org/eclipse/dataspaceconnector/azure/dataplane/azurestorage/pipeline/AzureStorageDataSource.java).  
```

> 11. The Provider DPF writes the data to the destination blob so that the consumer can access the data.
> The [AzureStorageDataSink](../../../../extensions/azure/data-plane/storage/src/main/java/org/eclipse/dataspaceconnector/azure/dataplane/azurestorage/pipeline/AzureStorageDataSink.java) transfers the data to the blob destination.

Reviewer: ```suggestion
11. The Provider DPF writes the data to the destination blob so that the consumer can access the data using an [AzureStorageDataSink](../../../../extensions/azure/data-plane/storage/src/main/java/org/eclipse/dataspaceconnector/azure/dataplane/azurestorage/pipeline/AzureStorageDataSink.java).
```

> 12. When the transfer is finished, the Provider DPF writes a blob called `.complete`.

Reviewer: ```suggestion
12. When the transfer is finished, the Provider DPF writes a blob called `.complete` to signal the completion.
```

> 13. In the meantime, the client polls the transfer status regularly on the consumer endpoint `/transferprocess/<PROCESS_ID>/state`.  

Reviewer: ```suggestion
```

> 14. In the meantime, the consumer regularly checks if a blob named `.complete` exists in the container.

Reviewer: ```suggestion
14. In the meantime, the consumer regularly checks if a blob named `.complete` exists in the container. Clients can poll the state using the `/transferprocess/<PROCESS_ID>/state` endpoint.
```

When the `.complete` is found, the consumer persists the new transferProcess state.  
15. When the transfer is finished, the client can read the blob.  

> 16. Then, the client can call the Data Management API to destroy the data.  

Reviewer: ```suggestion
16. Then, the client can call the Data Management API to deprovision the transfer process by the following two steps.
```

--> Make 17. and 18. substeps of 16.
